src/auth/computeclient.ts

```diff
@@ -83,7 +83,7 @@ export class Compute extends OAuth2Client {
       data = await gcpMetadata.instance({
         property: tokenPath,
         params: {
-          scopes: this.scopes,
+          scopes: this.scopes.join(','),
```

Comment:
The scopes are expected as a comma-separated string value, as opposed to the `?scope==scope1&scopes=scope2` style that `querystring.stringify()` uses: https://cloud.google.com/functions/docs/securing/function-identity#access_tokens

We could use `querystring.stringify(opts.params, {arrayFormat: 'comma'})`, but I'm not sure that every API endpoint has the same expected format when it comes to query string parameters with multiple values.